app/presenters/custom_buttons_mixin.rb

```diff
@@ -11,8 +11,9 @@ def buttons_ordered?(object)
   def get_custom_buttons(object)
     # FIXME: don't we have a method for the splits?
     # FIXME: cannot we ask for the null parent using Arel?
-    CustomButton.buttons_for(object.name.split('|').last.split('-').last).select do |uri|
-      uri.parent.nil?
+    button_name = object.name.split('|').last.split('-').last
+    CustomButton.buttons_for(button_name).includes(:custom_button_sets).select do |uri|
+      uri.custom_button_sets.blank?
```

Comment:
Again, for another PR or refactoring, I don't know what `uri` is here... it seems like a strange return object from `buttons_for`.  It's unclear what object type `uri` is.  From [here](https://github.com/ManageIQ/manageiq/blob/acdc1ccdaa3a01060415f2499455e3e3fe9b54ca/app/models/custom_button.rb#L161-L163), maybe `action_uri` would be more intention revealing to someone like me who's not familiar with this code.

Reply:
or even `resource_action` if that's what it is.